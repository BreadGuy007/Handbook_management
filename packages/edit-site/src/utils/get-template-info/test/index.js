/**
 * Internal dependencies
 */
import getTemplateInfo from '../';

describe( 'get template info', () => {
	it( 'should return an empty object if no template is passed', () => {
		expect( getTemplateInfo( null ) ).toEqual( {} );
		expect( getTemplateInfo( undefined ) ).toEqual( {} );
		expect( getTemplateInfo( false ) ).toEqual( {} );
	} );

	it( 'should return the default title if none is defined on the template', () => {
		expect( getTemplateInfo( { slug: 'index' } ).title ).toEqual( 'Index' );
	} );

	it( 'should return the rendered title if one is defined on the template', () => {
		expect(
			getTemplateInfo( {
				slug: 'index',
				title: { rendered: 'test title' },
			} ).title
		).toEqual( 'test title' );
	} );

	it( 'should return the slug if no title is found', () => {
		expect(
			getTemplateInfo( {
				slug: 'not a real template',
			} ).title
		).toEqual( 'not a real template' );
	} );

	it( 'should return the default description if none is defined on the template', () => {
		expect(
			getTemplateInfo( {
				slug: 'index',
			} ).description
		).toEqual( 'Default template' );
	} );

	it( 'should return the rendered excerpt as description if defined on the template', () => {
		expect(
			getTemplateInfo( {
				slug: 'index',
				excerpt: {
					rendered: 'test description',
				},
			} ).description
		).toEqual( 'test description' );
	} );

	it( 'should return both a title and a description', () => {
		expect(
			getTemplateInfo( {
				slug: 'index',
				excerpt: {
					rendered: 'test description',
				},
			} )
		).toMatchSnapshot();

		expect(
			getTemplateInfo( {
				slug: 'index',
			} )
		).toMatchSnapshot();

		expect( getTemplateInfo( {} ) ).toMatchSnapshot();
	} );
} );
