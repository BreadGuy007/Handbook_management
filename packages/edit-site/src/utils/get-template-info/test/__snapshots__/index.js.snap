// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get template info should return both a title and a description 1`] = `
Object {
  "description": "test description",
  "title": "Index",
}
`;

exports[`get template info should return both a title and a description 2`] = `
Object {
  "description": "Default template",
  "title": "Index",
}
`;

exports[`get template info should return both a title and a description 3`] = `
Object {
  "description": undefined,
  "title": undefined,
}
`;
